integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/massindexing/MassIndexingInheritanceIT.java

```diff
@@ -292,6 +376,95 @@ void multipleSameBranch() {
 		backendMock.verifyExpectationsMet();
 	}
 
+	@Test
+	void withMoreConcreteTypeDisabled() {
+		with( sessionFactory ).runNoTransaction( session -> {
+			SearchSession searchSession = Search.session( session );
+			MassIndexer indexer = searchSession.massIndexer( BA.class );
+
+
+			backendMock.expectWorks( BA.INDEX, DocumentCommitStrategy.NONE, DocumentRefreshStrategy.NONE )
+					.add( "100002", b -> b.field( "id", 100002L ) );
+			backendMock.expectWorks( BBA.INDEX, DocumentCommitStrategy.NONE, DocumentRefreshStrategy.NONE )
+					.add( "100004", b -> b.field( "id", 100004L ) );
+			backendMock.expectWorks( AAABA.INDEX, DocumentCommitStrategy.NONE, DocumentRefreshStrategy.NONE )
+					.add( "100005", b -> b.field( "id", 100005L ) );
+
+			backendMock.expectIndexScaleWorks( BA.INDEX )
+					.purge()
+					.mergeSegments()
+					.flush()
+					.refresh();
+			backendMock.expectIndexScaleWorks( BBA.INDEX )
+					.purge()
+					.mergeSegments()
+					.flush()
+					.refresh();
+			backendMock.expectIndexScaleWorks( AAABA.INDEX )
+					.purge()
+					.mergeSegments()
+					.flush()
+					.refresh();
+
+			try {
+				indexer.startAndWait();
+			}
+			catch (InterruptedException e) {
+				fail( "Unexpected InterruptedException: " + e.getMessage() );
+			}
+
+		} );
+
+		statementInspector.hasSelects( 9 )
+				//    select count(b1_0.id) from bba b1_0
+				//    select count(a1_0.id) from aaaba a1_0
+				//    select count(b1_0.id) from ba b1_0
+				//            join a b1_1 on b1_0.id=b1_1.id
+				//            left join aaaba b1_2 on b1_0.id=b1_2.id
+				//            left join aba b1_3 on b1_0.id=b1_3.id
+				//            left join bba b1_4 on b1_0.id=b1_4.id
+				//            where b1_1.type in (?)
+				.anyMatch( "select count\\([a-z0-9_.]+\\) from bba [a-z0-9_.]+" )
+				.anyMatch( "select count\\([a-z0-9_.]+\\) from aaaba [a-z0-9_.]+" )
+				.anyMatch( "select count\\([a-z0-9_.]+\\) from ba [a-z0-9_.]+ "
+						+ "join a [a-z0-9_.]+ on [a-z0-9_.=]+ "
+						+ "left join aaaba [a-z0-9_.]+ on [a-z0-9_.=]+ "
+						+ "left join aba [a-z0-9_.]+ on [a-z0-9_.=]+ "
+						+ "left join bba [a-z0-9_.]+ on [a-z0-9_.=]+ "
+						+ "where .+"
+				)
```

Comment:
from the looks of it ... ORM probably could do a bit better in this case, as all those left joins aren't really needed.

Reply:
Right... Probably something to request as an enhancement?

It's not *very* critical in our case as the query is only executed once, but it would be nice to get it solved.

Reply:
https://hibernate.atlassian.net/browse/HHH-17806